### DISCUSSION TITLE
Has anyone used prepackaged industrial Pi's instead kludging together HAT's?

### DISCUSSION THREAD

**Discussion by jamestalmage** [tagyoureit/nodejs-poolController#728] [Nixie]
It seems like there are a lot of interesting options out there. They bring benefits like pre-hardened I/O, wide temperature ranges, cleaner looking control box, and more compact form factors (bigger than a naked PI, but definitely smaller once you factor in all the exterior stuff you need to do to protect IO). Some even include integrated touch screens!

All of these do I/O a bit differently, but nothing seems non standard. I haven't dove into the njsPC codebase yet, but I'm assuming it's going to be pretty trivial to add profiles for the various inputs/relays these offer.

### [Factor 202](https://www.onlogic.com/fr202/)
- ~$900 with the integral touch screen
- -20˚ to 60˚C (-4˚F to 140F)
- 8-24VDC Power Input
- Optional PoE power module
- Optional 2.4" 320x240 touchscreen. 900 NIT (easily readable in direct sun)!!
- 4x Analog Inputs. Each one configurable as 0-5v, 0-10v, -5-5v, -10-10v, 0-20mA, 4-20mA or 4-24mA.
- 2x RTD Inputs. 2/3/4 wire
- 16 Digital In. Optically isolated. Optimized for 12-24V, active-high or active-low. Configurable internal pull-ups.
- 16 Digital Out. Optically isolated. Sourcing or sinking. Internal 0.5A resettable fuse. Up to 24V. Each bank of 8 outputs have isolated VCC/GND connections (can operate on voltages other than what the Pi does).
- [Manual Here](https://static.onlogic.com/resources/manuals/OnLogic-Fr202-Manual-V1.pdf)

### [EdgeLogix RPI-1000](https://www.seeedstudio.com/EdgeLogix-RPI-1000-CM4102032-p-5512.html) 
- ~$500 
- Includes 4.3" 480 x 800 touch screen
- -20˚ to 60˚C (-4˚F to 140F)
- 12 Digital In. Optically Isolated. 24V/1A rated
- 12 Digital Out. Optically Isolated. 24V/1A rated
- 5x RS48 Buses!! (Optically Isolated)
- 10.8-36V Input (I'm guessing DC only... But that's not specified)
- Standalone watch dog timer
- Wifi and BLE equipped
- Cellular and LoRa support via Mini-PCIe bus
- Relay modules are in development for "early 2023".

### [EdgeLogix RPI-200](https://www.seeedstudio.com/EdgeBox-RPi-200-CM4104016-p-5486.html)
- ~$250
- No screen
- 1x RS485
- 2x Digital In
- 2x Digital Out
- No analog ins (which kinda limits utility, but maybe if you set the heater via RS485 and just use that)

### [Industrial Shields Raspberry PLC](https://www.industrialshields.com/shop/product/012003000100-raspberry-plc-19r-4gb-2363)
- Too many configurations to list here... Link above is just to one. Suffice to say, it's got more I/O than you'll ever need
- Relay modules are rated to 220Vac & 5A / 30Vdc - 3A

**Comment by rstrouse** [comment]
I looked at these or something similar a few years ago.  While they seemed interesting, the thing that really throws sand in the works is that most of the time digital outputs are only normally open.  In the end you still will need relays to drive valves.   I also would not drive inductive loads off a 220vac 5A relay you should use a contactor for that.  Lights and other options will be fine but motors and even a large chlorinator power supply will eventually eat a relay.
